src/components/Image/index.js

```diff
@@ -7,67 +7,58 @@ import ONYXKEYS from '../../ONYXKEYS';
 import {defaultProps, imagePropTypes} from './imagePropTypes';
 import RESIZE_MODES from './resizeModes';
 
-class Image extends React.Component {
-    componentDidMount() {
-        this.configureOnLoad();
-    }
-
-    componentDidUpdate(prevProps) {
-        if (prevProps.source === this.props.source) {
-            return;
-        }
-        this.configureOnLoad();
-    }
-
+function Image(props) {
     /**
      * Check if the image source is a URL - if so the `encryptedAuthToken` is appended
      * to the source.
      * @returns {Object} - the configured image source
      */
-    getImageSource() {
-        const source = this.props.source;
+    const getImageSource = useCallback(() => {
+        const source = props.source;
```

Comment:
you can use earlier return like 
if (!isAuthTokenRequired) return source
            const authToken = lodashGet(props, 'session.encryptedAuthToken', null);
            return {uri: `${source.uri}?encryptedAuthToken=${encodeURIComponent(authToken)}`};